src/main/java/Duke.java

```diff
@@ -1,10 +1,323 @@
+import java.util.ArrayList;
+import java.io.IOException;
+import java.nio.file.FileAlreadyExistsException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Scanner;
+import duke.task.*;
+import duke.exception.*;
```

Comment:
Nice to see the use of new packages. However this is still explicit enough in terms of what classes will be used. See [https://se-education.org/guides/conventions/java/basic.html#package-and-import-statements](https://se-education.org/guides/conventions/java/basic.html#package-and-import-statements).

Reply:
thank you! i will explicitly import the classes